docs/lib-documentation.md

```diff
@@ -12,21 +12,22 @@ NetworkContext
 --------------
 This Kurtosis-provided class is the lowest-level representation of a test network, and provides methods for inspecting and manipulating the network. All [Network][network] implementations will encapsulate an instance of this class.
 
-### addServiceToPartition(ServiceID serviceId, PartitionID partitionId, [ContainerConfigFactory\<S\>][containerconfigfactory] configFactory) -\> ([ServiceContext][servicecontext] serviceContext, Map\<String, PortBinding\> hostPortBindings)
+### addServiceToPartition(ServiceID serviceId, PartitionID partitionId, [ContainerCreationConfig][containercreationconfig] containerCreationConfig, Func(String ipAddr, Map\<String, string\> generatedFileFilepaths, Map\<StaticFileID, string\> staticFileFilepaths) -\> ([ContainerRunConfig][containerrunconfig], error) generateRunConfigFunc) -\> ([ServiceContext][servicecontext] serviceContext, Map\<String, PortBinding\> hostPortBindings)
 Starts a new service in the network with the given service ID, inside the partition with the given ID, using the given config factory.
 
 **Args**
 
 * `serviceId`: The ID that the new service should have.
 * `partitionId`: The ID of the partition that the new service should be started in. This can be left blank to start the service in the default partition if it exists (i.e. if the network hasn't been repartitioned and the default partition removed).
-* `configFactory`: The factory that produces the configs Kurtosis will use to start the container for the new service.
+* `containerCreationConfig`: The definition of the necessary values that Kurtosis will use to start the container for the new service.
+* `generateRunConfigFunc`: An anonymous function which receives three dynamic values as arguments: the ip address, the generated file paths and the static file paths created during the starting process; and creates the ContainerRunConfig that Kurtosis will use to run the service
```

Comment:
This is quite good already! One suggested rephrase to clarify:
```
An anonymous function, used to produce the [ContainerRunConfig][containerrunconfig] for starting the service, which receives three dynamic values as arguments: the IP address of the service being started, the filepaths of the generated files specified in the [ContainerCreationConfig][containercreationconfig], and the filepaths of the static files specifid in [ContainerCreationConfig][containercreationconfig].
```

Reply:
I like it,  linking concepts,